extensions/integration-tests/src/schemaCompare.test.ts

```diff
@@ -0,0 +1,114 @@
+			assertSchemaCompareResult(schemaCompareResult);
+
+			let status = await service.schemaCompareGenerateScript(schemaCompareResult.operationId, 'testDb', path.join(__dirname, 'testScript_dacpac.sql'), azdata.TaskExecutionMode.execute);
+			assertScriptGenerationResult(status, path.join(__dirname, 'testScript_dacpac.sql'));
+		});
+
+		test('Schema compare database comparision and script generation', async function () {
+			let server = await getStandaloneServer();
+			await utils.connectToServer(server, 300);
+
+			let nodes = <azdata.objectexplorer.ObjectExplorerNode[]>await azdata.objectexplorer.getActiveConnectionNodes();
+			assert(nodes.length > 0, `Expecting at least one active connection, actual: ${nodes.length}`);
+
+			let index = nodes.findIndex(node => node.nodePath.includes(server.serverName));
+			assert(index !== -1, `Failed to find server: "${server.serverName}" in OE tree`);
+
+			let ownerUri = await azdata.connection.getUriForConnection(nodes[index].connectionId);
+
+			let dacfxService = await azdata.dataprotocol.getProvider<azdata.DacFxServicesProvider>('MSSQL', azdata.DataProviderType.DacFxServicesProvider);
+			assert(dacfxService, 'DacFx Service Provider is not available');
+
+			let now = new Date();
+			let result1 = await dacfxService.deployDacpac(path.join(__dirname, 'testData/Database1.dacpac'), 'database1_' + now.getTime().toString(), true, ownerUri, azdata.TaskExecutionMode.execute);
+			let result2 = await dacfxService.deployDacpac(path.join(__dirname, 'testData/Database2.dacpac'), 'database2_' + now.getTime().toString(), true, ownerUri, azdata.TaskExecutionMode.execute);
+
+			assert(result1.success === true, 'Deploy database 1 should succeed');
+			assert(result2.success === true, 'Deploy database 2 should succeed');
+
+			let service = await azdata.dataprotocol.getProvider<azdata.SchemaCompareServicesProvider>('MSSQL', azdata.DataProviderType.SchemaCompareServicesProvider);
+			assert(service, 'Schema Compare Service Provider is not available');
+
+			let source: azdata.SchemaCompareEndpointInfo = {
+				endpointType: azdata.SchemaCompareEndpointType.database,
+				packageFilePath: '',
+				databaseName: 'database1_' + now.getTime().toString(),
+				ownerUri: ownerUri,
+			};
+			let target: azdata.SchemaCompareEndpointInfo = {
+				endpointType: azdata.SchemaCompareEndpointType.database,
+				packageFilePath: '',
+				databaseName: 'database2_' + now.getTime().toString(),
+				ownerUri: ownerUri,
+			};
+
+			let schemaCompareResult = await service.schemaCompare(source, target, azdata.TaskExecutionMode.execute);
+			assertSchemaCompareResult(schemaCompareResult);
+
+			let status = await service.schemaCompareGenerateScript(schemaCompareResult.operationId, 'testDb1', path.join(__dirname, 'testScript_database.sql'), azdata.TaskExecutionMode.execute);
+			assertScriptGenerationResult(status, path.join(__dirname, 'testScript_database.sql'));
+
+			utils.deleteDB('database1_' + now.getTime().toString(), ownerUri);
```

Comment:
store the database name in a variable when it's first made since the time could be different here than when the db was created

Reply:
Now is precreated variable, so wouldn't be different. Added names at beginning for clarity though